tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -117,7 +126,11 @@ void test_wrap_alarm_instance(const char *dev_name)
 
 void test_wrap_alarm(void)
 {
+#ifdef CONFIG_COUNTER_MCUX_RTC
```

Comment:
it would be better to handle that in the test. If first call to `set_wrap` (now called `set_top_value`) returns `-ENOTSUP` test can stop the counter and leave without failure. We should avoid device specific defines in api test.